cartographer_grpc/framework/server_test.cc

```diff
@@ -173,6 +192,31 @@ TEST_F(ServerTest, ProcessBidiStreamingRpcTest) {
   server_->Shutdown();
 }
 
+TEST_F(ServerTest, WriteFromOtherThread) {
+  server_->SetExecutionContext(
+      cartographer::common::make_unique<MathServerContext>());
+  server_->Start();
+
+  proto::GetEchoResponse result;
+  proto::GetEchoRequest request;
+  request.set_input(13);
+
+  Server* server = server_.get();
+  std::thread response_thread([server]() {
+    std::future<EchoCallback> writer_future =
+        server->GetContext<MathServerContext>()->echo_writer_.get_future();
+    writer_future.wait();
+    CHECK(writer_future.get()());
```

Comment:
Assign the callback to a variable in a separate line so this becomes more readable.

Reply:
Done.